dev/com.ibm.ws.jaxws.2.2.webcontainer_fat_extended/test-applications/enableSchemaValidation/src/com/ibm/ws/jaxws/fat/mock/endpoint/MockJaxwsEndpointServlet.java

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.ws.jaxws.fat.mock.endpoint;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Logger;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This servlet acts as a Mock JAX-WS Endpoint that returns a "malformed" response that will cause the JAXB Unmarshaller on the
+ * client side to fail when processed.
+ *
+ * We take the inbound request, convert it to a string, and check for the "name" element, which will tell us which test is being invoked,
+ * based on that we will send a specific response, designed to cause a JAXB Unmarshaller failure.
+ *
+ * We set the publish servlet at http://127.0.0.1:8010/testHandlerProvider/SayHelloServiceWithHandler which
+ * allows the JAX-WS client in IgnoreUnexpectedElementTestServiceClient invoke this "Web Service".
+ *
+ * When enableSchemaValidation=false, these malformed responses may or may not still cause additional failures since the default schema validation will be disabled
+ * but the JAX-WS runtime still might not be able to process the message
+ */
+//Publishing the Servlet to the endpoint of a pre-existing Web Serbvice so we can use the client without modification
+@WebServlet("/SayHelloServiceWithHandler")
+public class MockJaxwsEndpointServlet extends HttpServlet {
+
+    Logger LOG = Logger.getLogger("MockJaxwsEndpointServlet.class");
+
+    /**
+     * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+     */
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        // Convert inbound request payload to a string
+        BufferedReader requestBufferedReader = request.getReader();
+        String requestPayload = new String();
+        for (String payLoadLine; (payLoadLine = requestBufferedReader.readLine()) != null; requestPayload += payLoadLine);
+
+        LOG.info("Received request: " + requestPayload);
+
+        // Now that we've got the request, we can the value of the "name" parameter sent by the client
+        // This will allow us to know which test we are running, and which response to send back to the client.
+        if (requestPayload.contains("AddedElement")) {
```

Comment:
I think you could refactor this if else statement to just set return value and then use the return value to set the response only once.